methods.py

```diff
@@ -1176,6 +1176,12 @@ def update_version():
     f.write("#define VERSION_STATUS " + str(version.status) + "\n")
     import datetime
     f.write("#define VERSION_YEAR " + str(datetime.datetime.now().year) + "\n")
+    head = open(".git/HEAD", "rb").readline().strip()
+    if head.startswith("ref: "):
+        ref = open(".git/" + head[5:], "rb").readline().strip()
+    else:
+        ref = head
```

Comment:
That's a bold assumption :) If building from a static archive without .git folder, nothing tells you that it's HEAD (also "head" in itself doesn't mean much, it's <branchname>/HEAD which points to the last commit of the branch). So a better default would be "unknown".

Still, as a distro packager, I can tell you I hate projects which do that :)
If there's no .git folder, just skip this logic and don't display the hash, that's much better.